docs/building-apps/build-properties.md

```diff
@@ -916,6 +916,85 @@ only scan libraries with the `[LinkWith]` attribute for Objective-C classes:
 </PropertyGroup>
 ```
 
+## RunWithOpen
+
+This property determines whether apps are launched using the `open` command on
+macOS, or if the app's executable is executed directly.
+
+This only applies to macOS and Mac Catalyst apps.
+
+The default value is `true`. In this mode, the app will be launched by macOS as any other UI application, any stdout/stderr output will be swallowed by macOS, and the `dotnet run` command will finish as soon as the app has launched.
+
+If set to `false`, any stdout/stderr output will be printed to the current terminal, and the `dotnet run` command won't finish until the app has exited.
+
+The following properties can be used to configure the behavior when set to `true` (i.e. using the `open` command):
+
+### OpenNewInstance
+
+If a new instance will be opened if the app is already running (defaults to `false`).
+
+This will pass `-n` to `open` if set to `true`.
+
+Example:
+
+```shell
+$ dotnet run -p:OpenNewInstance=false
+```
+
+### RunEnvironment
+
+This property can be used to set additional environment variables when the app is launched.
+
+Example:
+
+```shell
+$ dotnet run -p:RunEnvironment="--env VARIABLE1=VALUE1 --env VARIABLE2=VALUE2"
+```
+
+### StandardOutputPath
+
+This property can be used to redirect the stdout output from the app to a file.
+
+Example writing to a file:
+
+```shell
+$ dotnet run -p:StandardOutputPath=stdout.txt
+```
+
+Example writing to the current terminal:
+
+```shell
+$ dotnet run -p:StandardOutputPath=$(tty)
+[... Console.WriteLine output from app ...]
+```
+
+### StandardErrorPath
+
+This property can be used to redirect the stderr output from the app to a file.
+
+Example writing to a file:
+
+```shell
+$ dotnet run -p:StandardErrorPath=stderr.txt
+```
+
+Example writing to the current terminal:
+
+```shell
+$ dotnet run -p:StandardErrorPath=$(tty)
+[... Console.Error.WriteLine output from app ...]
+```
+
+### StandardInputPath
+
+This property can be used to redirect the stdin input to the app from a file.
+
+Example:
```

Comment:
Same here is this supported on all platforms sim and devices? Also if you are running on device is the file path provided local to the device/appbundle or will this be local to the machine that the device is connected via wifi/usb?  this might be obvious to us but just thinking about our customers and AI :P

Reply:
Same